File: server/app/routes/attendance.py
from fastapi import APIRouter, HTTPException
from app.database import employee_collection
from app.schemas import AttendanceCreate
from app.models import attendance_document

router = APIRouter()

@router.post("/attendance")
def mark_attendance(data: AttendanceCreate):
    employee = employee_collection.find_one(
        {"employee_id": data.employee_id}
    )

    if not employee:
        raise HTTPException(
            status_code=404,
            detail="Employee not found"
        )

File: client/src/services/attendance.js
import api from './api';

export const markAttendance = async (attendanceData) => {
  const response = await api.post('/attendance', attendanceData);
  return response.data;
};

export const getAttendance = async (employeeId) => {
  const response = await api.get(`/attendance/${employeeId}`);
  return response.data;
};
